plugins/fastem_roa_import_export.py

```diff
@@ -0,0 +1,341 @@
+        evt.Skip()
+        with wx.FileDialog(
+            self,
+            "Export ROAs to a CSV file",
+            wildcard="CSV files (*.csv)|*.csv",
+            style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
+        ) as fileDialog:
+            if fileDialog.ShowModal() == wx.ID_CANCEL:
+                return  # the user changed their mind
+
+            # save the current contents in the file
+            pathname = fileDialog.GetPath()
+            if not pathname.endswith(".csv"):
+                pathname += ".csv"
+            try:
+                with open(pathname, "w", newline="") as csvfile:
+                    data = {}
+                    writer = csv.DictWriter(csvfile, fieldnames=CSVFieldnames.list())
+                    writer.writeheader()
+                    project_ctrls = self._project_list_controller.project_ctrls
+                    for project_ctrl in project_ctrls.values():
+                        data[CSVFieldnames.PROJECT_NAME.value] = project_ctrl.model.name.value
+                        roas = project_ctrl.model.roas.value
+                        for roa in roas:
+                            data[CSVFieldnames.ROA_NAME.value] = roa.name.value
+                            roa_coordinates = roa.coordinates.value
+                            data[CSVFieldnames.ROA_COORDINATES_L.value] = roa_coordinates[0]
+                            data[CSVFieldnames.ROA_COORDINATES_T.value] = roa_coordinates[1]
+                            data[CSVFieldnames.ROA_COORDINATES_R.value] = roa_coordinates[2]
+                            data[CSVFieldnames.ROA_COORDINATES_B.value] = roa_coordinates[3]
+                            roc_2 = roa.roc_2.value
+                            data[CSVFieldnames.ROC_2_NAME.value] = roc_2.name.value
+                            roc_2_coordinates = roc_2.coordinates.value
+                            data[CSVFieldnames.ROC_2_COORDINATES_L.value] = roc_2_coordinates[0]
+                            data[CSVFieldnames.ROC_2_COORDINATES_T.value] = roc_2_coordinates[1]
+                            data[CSVFieldnames.ROC_2_COORDINATES_R.value] = roc_2_coordinates[2]
+                            data[CSVFieldnames.ROC_2_COORDINATES_B.value] = roc_2_coordinates[3]
+                            roc_3 = roa.roc_3.value
+                            data[CSVFieldnames.ROC_3_NAME.value] = roc_3.name.value
+                            roc_3_coordinates = roc_3.coordinates.value
+                            data[CSVFieldnames.ROC_3_COORDINATES_L.value] = roc_3_coordinates[0]
+                            data[CSVFieldnames.ROC_3_COORDINATES_T.value] = roc_3_coordinates[1]
+                            data[CSVFieldnames.ROC_3_COORDINATES_R.value] = roc_3_coordinates[2]
+                            data[CSVFieldnames.ROC_3_COORDINATES_B.value] = roc_3_coordinates[3]
+                            writer.writerow(data)
+            except Exception as ex:
+                logging.exception(ex)
```

Comment:
`logging.exception` already shows the exception, so no need to pass `ex`. You can add a text more like "Failure when exporting RoAs".